Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp

```diff
@@ -607,35 +607,78 @@ u8 iso_days_in_month(i32 year, u8 month)
 // 12.2.31 ToISOWeekOfYear ( year, month, day ), https://tc39.es/proposal-temporal/#sec-temporal-toisoweekofyear
 u8 to_iso_week_of_year(i32 year, u8 month, u8 day)
 {
-    // 1. Assert: year is an integer.
-    // 2. Assert: month is an integer.
-    // 3. Assert: day is an integer.
-
-    // 4. Let date be the date given by year, month, and day.
-    // 5. Return date's week number according to ISO-8601 as an integer.
-    auto t = make_date(make_day(year, month - 1, day), 0);
-    auto day_of_year = day_within_year(t) + 1;
-    auto day_of_week = week_day(t);
-    if (day_of_week == 0)
-        day_of_week = 7;
-    auto week = (day_of_year - day_of_week + 10) / 7;
+    // 1. Assert: IsValidISODate(year, month, day) is true.
+    VERIFY(is_valid_iso_date(year, month, day));
+
+    // 2. Let wednesday be 3.
+    constexpr auto wednesday = 3;
+
+    // 3. Let thursday be 4.
+    constexpr auto thursday = 4;
+
+    // 4. Let friday be 5.
+    constexpr auto friday = 5;
+
+    // 5. Let saturday be 6.
+    constexpr auto saturday = 6;
+
+    // 6. Let daysInWeek be 7.
+    constexpr auto days_in_week = 7;
+
+    // 7. Let maxWeekNumber be 53.
+    constexpr auto max_week_number = 53;
+
+    // 8. Let dayOfYear be ToISODayOfYear(year, month, day).
+    auto day_of_year = to_iso_day_of_year(year, month, day);
+
+    // 9. Let dayOfWeek be ToISODayOfWeek(year, month, day).
+    auto day_of_week = to_iso_day_of_week(year, month, day);
 
+    // 10. Let week be floor((dayOfYear + daysInWeek - dayOfWeek + wednesday ) / daysInWeek).
+    auto week = (day_of_year + days_in_week - day_of_week + wednesday) / days_in_week;
```

Comment:
This seems to be missing `floor`, is it not required here?

Reply:
It _should_ do the same using rounding from integer math, but I guess that's not true for negative `week` results - will fix :+1: